### PR TITLE
JetBrains Runtime Update

### PR DESCRIPTION
**Pull request by dan-smetana**
## Description of changes

This updates the JetBrains runtime, enabling client-side decorations for JetBrains IDEs among other things.

## Things done

Updates the JetBrains runtime. A small change had to be made when building JCEF to include its version information as that is now required by the runtime.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/jetbrains-jdk/default.nix [hunk 1]
```diff
@@ -64,8 +64,8 @@ openjdk17.overrideAttrs (oldAttrs: rec {
   buildPhase = ''
     runHook preBuild
 
-    mkdir -p jcef_linux_${arch}/jmods
-    cp ${jetbrains.jcef}/* jcef_linux_${arch}/jmods
+    mkdir -p jcef_linux_${arch}/
+    cp -r ${jetbrains.jcef}/* jcef_linux_${arch}/
```
Comment (Mic92) [inline pkgs/development/compilers/jetbrains-jdk/default.nix:68]:
```suggestion
    cp -r --reflink=auto ${jetbrains.jcef}/* jcef_linux_${arch}/
```

**Comment by Mic92** [comment]
Does https://github.com/NixOS/nixpkgs/pull/265558 depend on this?

File: pkgs/development/compilers/jetbrains-jdk/default.nix (hunk 1 above)
Comment (danielslee) [reply pkgs/development/compilers/jetbrains-jdk/default.nix:68]: Isn't reflink auto the default now? 

>   cp and install now default to copy-on-write (COW) if available.

https://lists.gnu.org/archive/html/coreutils/2021-09/msg00113.html

**Comment by danielslee** [comment]
> Does #265558 depend on this?

No, you just won't get client-side decorations without it. But it does work. I just tried it.

**Comment by wegank** [comment]
Merge conflict, please rebase.

**Comment by MattSturgeon** [comment]
Sounds like this fixes #272892 ?

**Comment by liff** [comment]
This is obsolete as the runtime was recently updated to 21 in #318036.

**Pull request closed without merging**